internal/validations/validations.go

```diff
@@ -111,7 +111,7 @@ func isValidImageURI(imageURI string) bool {
 }
 
 func isValidGitRepository(fl validator.FieldLevel) bool {
-	return regexp.MustCompile(`^git@([\w\d\.]+):(.*)`).MatchString(fl.Field().String())
+	return regexp.MustCompile(`((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`).MatchString(fl.Field().String())
```

Comment:
i don't think we support `http` (per [this](https://github.com/argoproj-labs/argo-cloudops/pull/54/files#diff-2d5cb73ac2d545d65540131e83a6b9fe69b84c37c5453e94d71046f572a9c50dR51-R57))? Or maybe we label it `https`, but `http` does work? I'd be totally fine with only `https` and not enabling sending creds in clear text.

Reply:
it looks like technically http does work if i try a repo with http though i think github might be autoredirecting to https on their end. i just labelled the function https cause i honestly forgot about http. we can change the method to http since the only difference otherwise is the url that the user sets

Reply:
I think I'd rather just drop `http` (so `...(git|ssh|https)...`from this regex so we don't let people do something insecurely with the creds at this point.

Reply:
sounds good to me